src/ol_infrastructure/lib/consul.py

```diff
@@ -0,0 +1,49 @@
+import base64
+import json
+import textwrap
+import yaml
+
+
+def build_consul_userdata(env_name):
+    """ Builds consul userdata for cloud init, to create auto scale group and launch configs
+        TODO setup for consul_esm """
+    return base64.b64encode(
+        "#cloud-config\n{}".format(
+            yaml.dump(
+                {
+                    "write_files": [
+                        {
+                            "path": "/etc/consul.d/99-autojoin.json",
```

Comment:
For Consul servers, we will also need a way to pass in a list of datacenters that they will peer with across VPCs. That will go into a `retry_join_wan` configuration parameter. Would probably make sense to keep that as a separate file for simplifity (e.g. 99-autojoin-wan.json) (https://www.consul.io/docs/agent/options#_retry_join_wan)